src/server/pps/server/api_server.go

```diff
@@ -1494,6 +1495,19 @@ func (a *apiServer) validateEnterpriseChecks(ctx context.Context, req *pps.Creat
 	return nil
 }
 
+func (a *apiServer) validateSecret(ctx context.Context, req *pps.CreatePipelineRequest) error {
+	for _, s := range req.GetTransform().GetSecrets() {
+		_, err := a.env.KubeClient.CoreV1().Secrets(a.namespace).Get(ctx, s.Name, metav1.GetOptions{})
```

Comment:
should we check `s.Key` exists, too? Not sure we're guaranteed to have access to that, I guess

Reply:
oh, I actually meant in the secret object returned by k8s as well, but I don't really know enough to say if that makes sense

Reply:
Ah, I think that _does_ make sense. Implemented that as well.